modules/dr-recover-lost-control-plane-hosts.adoc

```diff
@@ -0,0 +1,301 @@
++
+If your load balancing is managed by AWS, see link:https://docs.aws.amazon.com/elasticloadbalancing/latest/application/target-group-register-targets.html#register-ip-addresses[Register or Deregister Targets by IP Address] in the AWS documentation for instructions on updating load balancer entries.
+
+. Grow etcd to full membership.
+
+.. Set up a temporary etcd certificate signer service on your master where you have restored etcd.
+
+... Access the original master, and log in to your cluster as a `cluster-admin` user using the following command.
++
+----
+[core@ip-10-0-143-125 ~]$ oc login https://localhost:6443
+Authentication required for https://localhost:6443 (openshift)
+Username: kubeadmin
+Password:
+Login successful.
+----
+
+... Obtain a local copy of the pull secret used to install the cluster.
++
+----
+[core@ip-10-0-143-125 ~]$ oc get secret pull-secret -n openshift-config -o jsonpath={.data.'\.dockerconfigjson'} | base64 -d > config.json
+----
+
+... Obtain the pull specification for the `kube-etcd-signer-server` image.
++
+----
+[core@ip-10-0-143-125 ~]$ export KUBE_ETCD_SIGNER_SERVER=$(oc adm release info --image-for kube-etcd-signer-server --registry-config=./config.json)
+----
+
+... Run the `tokenize-signer.sh` script.
++
+Be sure to pass in the `-E` flag to `sudo` so that environment variables are properly passed to the script.
++
+----
+[core@ip-10-0-143-125 ~]$ sudo -E /usr/local/bin/tokenize-signer.sh ip-10-0-143-125 <1>
+Populating template /usr/local/share/openshift-recovery/template/kube-etcd-cert-signer.yaml.template
+Populating template ./assets/tmp/kube-etcd-cert-signer.yaml.stage1
+Tokenized template now ready: ./assets/manifests/kube-etcd-cert-signer.yaml
+----
+<1> The host name of the original master you just restored, where the signer should be deployed.
+
+... Create the signer Pod using the file that was generated.
++
+----
+[core@ip-10-0-143-125 ~]$ oc create -f assets/manifests/kube-etcd-cert-signer.yaml
+pod/etcd-signer created
+----
+
+... Verify that the signer is listening on this master node.
++
+----
+[core@ip-10-0-143-125 ~]$ ss -ltn | grep 9943
+LISTEN   0         128                       *:9943                   *:*
+----
+
+.. Add the new master hosts to the etcd cluster.
++
+// TODO: Check accuracy of this ^.
+
+... Access one of the new master hosts, and log in to your cluster as a `cluster-admin` user using the following command.
++
+----
+[core@ip-10-0-156-255 ~]$ oc login https://localhost:6443
+Authentication required for https://localhost:6443 (openshift)
+Username: kubeadmin
+Password:
+Login successful.
+----
+
+... Obtain a local copy of the pull secret used to install the cluster.
++
+----
+[core@ip-10-0-156-255 ~]$ oc get secret pull-secret -n openshift-config -o jsonpath={.data.'\.dockerconfigjson'} | base64 -d > config.json
+----
```

Comment:
same as above this can be removed.

--registry-config='/var/lib/kubelet/config.json'